examples/result_file_dataclasses.py

```diff
@@ -0,0 +1,202 @@
+"""
+Working with result files using Python-native dataclasses.
+"""
+from collections.abc import Iterator
+from operator import attrgetter
+from pathlib import Path
+
+import pendulum
```

Comment:
if we're not adding polars / pendulum as dependencies to the repo, I'd consider removing those from the examples file (or commenting that these are optional)

Reply:
I've removed these from the examples, opting to use a simpler string-based normalization rule and native CSV writer instead.